piet-gpu/src/samples.rs

```diff
@@ -101,63 +101,60 @@ pub fn render_scene(sb: &mut SceneBuilder) {
 #[allow(unused)]
 fn render_cardioid(sb: &mut SceneBuilder) {
     let n = 601;
-    let dth = std::f32::consts::PI * 2.0 / (n as f32);
+    let dth = std::f64::consts::PI * 2.0 / (n as f64);
     let center = Point::new(1024.0, 768.0);
     let r = 750.0;
     let mut path = vec![];
     for i in 1..n {
         let mut p0 = center;
-        let a0 = i as f32 * dth;
+        let a0 = i as f64 * dth;
         p0.x += a0.cos() * r;
         p0.y += a0.sin() * r;
         let mut p1 = center;
-        let a1 = ((i * 2) % n) as f32 * dth;
+        let a1 = ((i * 2) % n) as f64 * dth;
         p1.x += a1.cos() * r;
         p1.y += a1.sin() * r;
-        path.push(PathElement::MoveTo(p0));
-        path.push(PathElement::LineTo(p1));
+        path.push(PathEl::MoveTo(p0));
+        path.push(PathEl::LineTo(p1));
     }
     sb.stroke(
-        &simple_stroke(2.0),
+        &Stroke::new(2.0),
         Affine::IDENTITY,
-        &Brush::Solid(Color::rgb8(0, 0, 0)),
+        Color::rgb8(0, 0, 0),
         None,
-        &path,
+        &&path[..],
```

Comment:
We can impl Shape for `Vec<PathEl>` in kurbo to fix this.

Reply:
Hmmm. The way I've dealt with this in the path is use `BezPath` which is basically just a newtype for that, and I think I wanted to encourage people to use that rather than the raw vec. So I think the most idiomatic way to write this in this function is `path = BezPath::new()`.

I don't want to rathole on this too much, but I do find that sample code often has a disproportionate impact, as people often use it as a starting point.

Reply:
Fair enough! I believe it’s just a drop in replacement anyway.